pkg/interactive/interactive.go

```diff
@@ -0,0 +1,134 @@
+package interactive
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/kit/fs"
+	"github.com/kolide/launcher/pkg/osquery/table"
+	osquery "github.com/osquery/osquery-go"
+)
+
+const extensionName = "com.kolide.launcher_interactive"
+
+func StartProcess(rootDir, osquerydPath string, osqueryFlags []string) (*os.Process, error) {
+
+	if err := os.MkdirAll(rootDir, fs.DirMode); err != nil {
+		return nil, fmt.Errorf("creating root dir for interactive mode: %w", err)
+	}
+
+	socketPath, err := socketPath(rootDir, osqueryFlags)
+	if err != nil {
+		return nil, fmt.Errorf("creating socket path: %w", err)
+	}
+
+	// Get the current working directory.
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("error getting current working directory: %w", err)
+	}
+
+	// Transfer stdin, stdout, and stderr to the new process
+	// and also set target directory for the shell to start in.
+	pa := os.ProcAttr{
+		Files: []*os.File{os.Stdin, os.Stdout, os.Stderr},
+		Dir:   cwd,
+	}
+
+	proc, err := os.StartProcess(osquerydPath, buildOsqueryFlags(socketPath, osqueryFlags), &pa)
+	if err != nil {
+		return nil, fmt.Errorf("error starting osqueryd in interactive mode: %w", err)
+	}
+
+	_, err = loadExtensions(socketPath, osquerydPath)
+	if err != nil {
+		return nil, fmt.Errorf("error loading extensions: %s", err)
+	}
+
+	return proc, nil
+}
+
+func buildOsqueryFlags(socketPath string, osqueryFlags []string) []string {
+
+	// putting "-S" (the interactive flag) first because the behavior is inconsistent
+	// when it's in the middle, found this during development on M1 macOS monterey 12.4
+	// ~James Pickett 07/05/2022
+	flags := []string{"-S"}
+
+	for _, flag := range osqueryFlags {
+		flags = append(flags, fmt.Sprintf("--%s", flag))
+	}
+
+	// required flags for interactive mode to function
+	flags = append(flags, []string{
+		"--disable_extensions=false",
+		"--extensions_timeout=10",
+		fmt.Sprintf("--extensions_require=%s", extensionName),
+		fmt.Sprintf("--extensions_socket=%s", socketPath),
+	}...)
+
+	return flags
+}
+
+// MaxSocketPathCharacters is set to 97 because a ".12345" uuid is added to the socket down stream
+// if the provided socket is greater than 97 we may exceed the limit of 103
+// why 103 limit? https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars
+const MaxSocketPathCharacters = 97
+
+func socketPath(rootDir string, osqueryFlags []string) (string, error) {
+
+	path := ""
+
+	for _, flag := range osqueryFlags {
+		if strings.HasPrefix(flag, "extensions_socket=") {
+			split := strings.Split(flag, "=")
+			if len(split) > 1 && split[1] != "" {
+				path = split[1]
+			} else {
+				return "", errors.New("extensions_socket flag is missing a value")
+			}
+		}
+	}
+
+	if path == "" {
+		path = filepath.Join(rootDir, "sock")
```

Comment:
```suggestion
		path = filepath.Join(rootDir, "osquery.sock")
```

Unless we're length restricted, slight preference for more descriptive

Reply:
`osquery.sock` is a bit tight for running unit tests:

```
error loading extensions: error creating extension manager server: socket path /var/folders/8p/40gfkn2d3zl3mb_m9jk_vsk80000gn/T/TestStartProcessno_flags978291810/001/osquery.sock (99 characters) exceeded the maximum socket path character length of 97
```

how about `osq.sock` ?

Bring us down to 95 characters and from what I've seen the test dir only vary by 1-2 characters